.github/workflows/release.yaml

```diff
@@ -0,0 +1,88 @@
+name: Release a new version
+
+on:
+  release:
+    types:
+      - published
+
+env:
+  NODE_OPTIONS: '--max-old-space-size=8192'
+
+jobs:
+  publish-to-npm:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check workflow runs in a release branch
+        run: |
+          if [[ "${{ github.event.release.target_commitish }}" != *releases* ]]; then
+            echo "Expecting to be in a release branch, but we are in: ${{ github.event.release.target_commitish }}"
+            exit 1
+          fi
+
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.PUSH_TO_PROTECTED_BRANCH }}
```

Comment:
We don't have this secret in the settings ATM.

Reply:
I removed this one and instead used 
```
permissions:
  contents: write
```

hard to test though :)

Reply:
Yes, we'll need to do an actual release to test it for real.
But the change looks good.